### PR TITLE
Issue #2735441: Port commerce_order_handler_area_order_total

### PR DESCRIPTION
**Pull request by geertvd**

### CODE REVIEW

File: modules/order/src/Plugin/views/area/OrderTotal.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\commerce_order\Plugin\views\area\OrderTotal.
```
Comment (mglaman) [inline modules/order/src/Plugin/views/area/OrderTotal.php]:
```
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 2 | ERROR | [x] Namespaced classes, interfaces and traits should not
   |       |     begin with a file doc comment
   |       |     (Drupal.Commenting.FileComment.NamespaceNoFileDoc)
```

**Comment by mglaman** [comment]
@bojanz looks good, just needs rebase/squash.

File: modules/order/commerce_order.views.inc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * @file
+ * Provide views data that isn't tied to any other module.
+ */
+
+/**
+ * Implements hook_views_data().
+ */
+function commerce_order_views_data() {
+  $data['views']['commerce_order_total'] = array(
```
Comment (bojanz) [inline modules/order/commerce_order.views.inc]: We don't use array( in code anymore, needs to be the short array syntax: [ ]

File: modules/order/src/Plugin/views/area/OrderTotal.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\views\Plugin\views\area\AreaPluginBase;
+use Drupal\views\Plugin\views\argument\NumericArgument;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines an order total area handler.
+ *
+ * Defines an order total area handler that shows the order total field with its
```
Comment (bojanz) [inline modules/order/src/Plugin/views/area/OrderTotal.php]: "Defines an order total area handler that" is redudant, already written in the line above. Start with "Shows the order total...'

File: modules/order/src/Plugin/views/area/OrderTotal.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    $form['empty']['#description'] = $this->t("Even if selected, this area handler will never render if a valid order cannot be found in the View's arguments.");
```
Comment (bojanz) [inline modules/order/src/Plugin/views/area/OrderTotal.php:66]: Do we need this description? Technically the entire view should not render if there is no valid argument?

**Comment by geertvd** [comment]
> Do we need this description? Technically the entire view should not render if there is no valid argument?

I'd leave that one in, the area handler will still be available for placement in other views and could use an explanation why it won't be rendered.

Will take care of the other feedback later today

File: modules/order/commerce_order.views.inc [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * @file
+ * Provide views data that isn't tied to any other module.
```
Comment (bojanz) [inline modules/order/commerce_order.views.inc:5]: That's a weird comment. Why not "Provides views data for the Order module." or "Provides views data for orders."?

**Comment by mglaman** [comment]
Talked with @bojanz, going to address tweak, squash, and merge. Will update with the final commit hash.

**Comment by mglaman** [comment]
Committed as @3014f23d171190bdb92ebf24fbf7adb579f833a8

**Pull request closed without merging**